### PR TITLE
completed project

### PR DESCRIPTION
**Pull request by hsiupinggay**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment? **2-3**

Please fill in one error and/or error message you received while working on this assignment. **unknown {**

What part of the assignment did you spend the most time on? **Figuring out how to have the output displayed efficiently**

Comfort Level (1-5): **4**

Completeness Level (1-5): **If this means how completed my project was then I'll say 4.5.**

What did you think of this deliverable? **I liked it, I had fun. The logic of the game wasn't too difficult to get after all that we have learnt in class, I felt well-prepared for it, so it was nice to spend time thinking about nomenclature, and making the output look nice.**

Is there anything in this code that you feel pleased about? **It's easy to build on once I got the core logic right, the ease to build on the "reversed" game logic was pretty satisfying. I also enjoyed the part about making the output look nice**

### CODE REVIEW

**Comment by hsiupinggay** [comment]
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment? 2-3

Please fill in one error and/or error message you received while working on this assignment. undefined

What part of the assignment did you spend the most time on? Refactoring the code after adding modes

Comfort Level (1-5): 4

Completeness Level (1-5): 3.5

What did you think of this deliverable? It was quite challenging refactoring the code, but it helped me understand the asynchrony of the language and program.

Is there anything in this code that you feel pleased about? Getting it to work

File: script.js [hunk 1]
```diff
@@ -1,4 +1,195 @@
...
+  }
+
+  // performance metrics
+  totalNonDrawRounds = winCount + lossCount;
+  if (winCount > lossCount) {
+    var playerPerformance = `So far ${userName}, you've won ${winCount}/${totalNonDrawRounds} rounds that are not draws, keep it up!`;
+  } else if (winCount < lossCount) {
+    playerPerformance = `So far ${userName}, you've won ${winCount}/${totalNonDrawRounds} rounds that are not draws, don't be too disheartened.`;
+  } else if ((winCount = lossCount)) {
+    playerPerformance = `So, ${userName}, you've been winning the same number of times as the computer, see if you can tip the scales in your favour.`;
```
Comment (zephaniahong) [inline script.js:88]: Consider abstracting this into a separate function such as checkPerformance as it seems like you repeated this code below. A good tell tale sign of when to create a function is when you repeat the same code more than once

File: script.js [hunk 2]
```diff
@@ -1,4 +1,195 @@
...
+    (input == "paper" && programPlays == "stone") ||
+    (input == "stone" && programPlays == "scissors")
+  ) {
+    gameOutcomeOutput = `You lose！ Bummer. 
+    <br> You've won ${winCount} time(s), the computer has won ${lossCount} time(s)`;
+  } else if (
+    (input == "scissors" && programPlays == "stone") ||
+    (input == "paper" && programPlays == "scissors") ||
+    (input == "stone" && programPlays == "paper")
+  ) {
```
Comment (zephaniahong) [inline script.js:129]: This could be abstracted into a checkWinner function as this logic is also repeated several time

File: script.js [hunk 3]
```diff
@@ -1,4 +1,195 @@
...
+  var programPlays;
+  if (integerSelected == 0) {
+    programPlays = "scissors";
+  } else if (integerSelected == 1) {
+    programPlays = "paper";
+  } else if (integerSelected == 2) {
+    programPlays = "stone";
+  }
+  return programPlays;
+}
```
Comment (zephaniahong) [inline script.js:190]: Good use and naming of function.

**Review by zephaniahong — COMMENTED** [review]
Overall, a very good job with this project! You have a strong understanding of functions, conditional statements and booleans. Continue using functions the way you are using them. An improvement would be that once you are done with the project, take some time to see which parts of the code are being repeated and consider abstracting them into helper functions to make the overall code even cleaner. But really good job with keep the main function very logical and clean

**Pull request closed without merging**